### PR TITLE
Add user setting for toggle typing privacy

### PR DESCRIPTION
**Pull request by vitormattos**
API side of #9248 to implement an user setting to toggle typing privacy

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: lib/Config.php [hunk 1]
```diff
@@ -87,6 +87,14 @@ public function getUserReadPrivacy(string $userId): int {
 			(string) Participant::PRIVACY_PUBLIC);
 	}
 
+	public function getTypingPrivacy(string $userId): int {
```
Comment (nickvergessen) [inline lib/Config.php:90]:
```suggestion
	public function getUserTypingPrivacy(string $userId): int {
```

**Review by nickvergessen — CHANGES_REQUESTED** [review]
* Missing from the list in https://github.com/nextcloud/spreed/blob/70e25dc91c16f1f95a58237848020a0f74166772/docs/settings.md#L27
* Missing in https://github.com/nextcloud/spreed/blob/5ac5d00e2d64f580aadc3257d166c1df4c904313/lib/Controller/SettingsController.php

Please also search if the preference name, capability name or config function is used anywhere else

File: docs/constants.md [hunk 2]
```diff
@@ -76,6 +76,10 @@
 * `0` Read status is public
 * `1` Read status is private
 
+### Participant typing privacy
+* `0` Make visible the typing status
+* `1` Hide the typing status
```
Comment (nickvergessen) [inline docs/constants.md:81]:
```suggestion
* `0` Typing status is public
* `1` Typing status is private
```

File: lib/Controller/AvatarController.php
Comment (nickvergessen) [inline lib/Controller/AvatarController.php:1]: https://github.com/nextcloud/spreed/pull/9459

File: lib/Controller/SettingsController.php [hunk 3]
```diff
@@ -114,6 +114,10 @@ protected function validateUserSetting(string $setting, $value): bool {
 			return false;
 		}
 
+		if ($setting === 'typing_privacy') {
+			return $value === '1' || $value === '0';
+		}
+
 		if ($setting === 'read_status_privacy') {
```
Comment (nickvergessen) [inline lib/Controller/SettingsController.php:121]:
```suggestion
		if ($setting === 'typing_privacy' || $setting === 'read_status_privacy') {
```

File: lib/TInitialState.php [hunk 4]
```diff
@@ -190,6 +195,11 @@ protected function publishInitialStateForGuest(): void {
 			Participant::PRIVACY_PUBLIC
 		);
 
+		$this->initialState->provideInitialState(
+			'typing_privacy',
+			'0'
```
Comment (nickvergessen) [inline lib/TInitialState.php:200]:
```suggestion
			Participant::PRIVACY_PUBLIC
```

File: tests/integration/features/chat-2/typing-privacy.feature [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Feature: chat-2/typing-privacy
+  Background:
+    Given user "participant1" exists
+  Scenario: User toggles the typing privacy
+    # Hide
```
Comment (nickvergessen) [inline tests/integration/features/chat-2/typing-privacy.feature:5]:
```suggestion
    # Private
```

File: tests/integration/features/chat-2/typing-privacy.feature [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    When user "participant1" sets setting "typing_privacy" to "1" with 200 (v1)
+    Then user "participant1" has capability "spreed=>config=>chat=>typing-privacy" set to "1"
+
+    # Visible
```
Comment (nickvergessen) [inline tests/integration/features/chat-2/typing-privacy.feature:9]:
```suggestion
    # Public
```

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Review by nickvergessen — APPROVED** [review]

**Comment by SystemKeeper** [comment]
Fine by me.
CC: @danxuliu About the capability name, so you're aware

**Pull request merged**